src/test/scala/dsptools/VerboseDspTesterSpec/SimpleTBwGenTypeOption.scala

```diff
@@ -0,0 +1,401 @@
+          isVerbose = true)
+      testerOptions = testerOptionsGlobal
+    }
+
+  val optionsFail = new VerboseDspTesterOptionsManager {
+    verboseDspTesterOptions = optionsPass.verboseDspTesterOptions.copy(fixTolLSBs = 0)
+    testerOptions = testerOptionsGlobal
+  }
+
+  val optionsPassTB = new VerboseDspTesterOptionsManager {
+    verboseDspTesterOptions = optionsPass.verboseDspTesterOptions.copy(genVerilogTb = true)
+    testerOptions = testerOptionsGlobal
+  }
+
+}
+
+class SimpleTBSpec extends FlatSpec with Matchers {
+
+  val p = TestSetup.p
+  val optionsPass = TestSetup.optionsPass
+  val optionsPassTB = TestSetup.optionsPassTB
+  val optionsFail = TestSetup.optionsFail
+
+  behavior of "simple module lits"
+
+  it should "properly read lits with gen = sint (reals rounded) and expect tolerance set to 1 bit" in {
+    dsptools.Driver.execute(() => new SimpleLitModule(p.genShortS, p.genLongS, includeR = true, p), optionsPass) { c =>
+      new PassLitTester(c)
+    } should be (true)
+  }
+
+  it should "properly read lits with gen = fixed and expect tolerance set to 1 bit " +
+      "(even with finite fractional bits)" in {
+    dsptools.Driver.execute(() => new SimpleLitModule(p.genShortF, p.genLongF, includeR = true, p), optionsPass) { c =>
+      new PassLitTester(c)
+    } should be (true)
+  }
+
+  it should "*fail* to read all lits with gen = fixed when expect tolerance is set to 0 bits " + 
+      "(due to not having enough fractional bits to represent #s)" in {
+    dsptools.Driver.execute(() => new SimpleLitModule(p.genShortF, p.genLongF, includeR = true, p), optionsFail) { c =>
+      new FailLitTester(c)
+    } should be (false)
+  }
+
+  behavior of "simple module registered io"
+
+  it should "properly poke/peek io delayed 1 cycle with gen = sint (reals rounded) " +
+      "and expect tolerance set to 1 bit" in {
+    dsptools.Driver.execute(() => new SimpleIOModule(p.genShortS, p.genLongS, includeR = true, p), optionsPass) { c =>
+      new PassIOTester(c)
+    } should be (true)
+  }
+
+  it should "properly poke/peek io delayed 1 cycle with gen = fixed and expect tolerance set to 1 bit" in {
+    dsptools.Driver.execute(() => new SimpleIOModule(p.genShortF, p.genLongF, includeR = true, p), optionsPass) { c =>
+      new PassIOTester(c)
+    } should be (true)
+  }
+
+  it should "properly poke/peek io delayed 1 cycle with gen = fixed + print TB (no reals)" in {
+    this.synchronized {
+      var tbFileLoc: String = ""
+      dsptools.Driver.execute(() => new SimpleIOModule(p.genShortF, p.genLongF, includeR = false, p), 
+          optionsPassTB) { c => {
+        val tester = new PassIOTester(c)
+        tbFileLoc = tester.tbFileName
+        tester
+      } } should be (true)
+      def read(file: String) = scala.io.Source.fromFile(file).getLines.mkString("\n")
+      val tbTxt = read(tbFileLoc).stripMargin
+      require(tbTxt == TbGoldenModel.txt, "Test bench incorrect! (compared against simulated model)")
```

Comment:
Sure thing, it was a nitpick. Did you look at putting it into src/test/resources? It would make it easier to e.g. make sure a test case works with some tool